sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -123,6 +153,18 @@ case class InSet(child: Expression, hset: Set[Any])
   override def eval(input: InternalRow): Any = {
     hset.contains(child.eval(input))
   }
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val setName = classOf[Set[Any]].getName
+    val childGen = child.gen(ctx)
+    ctx.references += hset
```

Comment:
Can we still push `this` into `ctx.references`, using `expressions[${ctx.references.size - 1}].hset` to access it?

If the accessor is heavy, we could cache it using ctx.addMutableState.